code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx

```diff
@@ -220,9 +220,14 @@ export class PreviewWithSelection<TFramework extends Renderer> extends Preview<T
   }
 
   async onSetCurrentStory(selection: { storyId: StoryId; viewMode?: ViewMode }) {
+    /**
+     * At the end of the initialization promise we will read the current story from the selection store,
+     * so make sure we've updated it with the new selection or we'll lose track of it at the end of init.
+     */
+    this.selectionStore.setSelection({ viewMode: 'story', ...selection });
+
     await this.storyStore.initializationPromise;
```

Comment:
I'd still like a unit test for this change

Reply:
added!